coreos-installer

```diff
@@ -591,6 +591,18 @@ write_image_to_disk() {
     udevadm settle
 }
 
+############################################################
+#Make call home HTTP(s) call as the final step
+############################################################
+call_home() {
+    CALLHOME_URL=$(cat /tmp/callhome_url)
+    if [ ! -z "$CALLHOME_URL" ]
+      echo "Calling home: $CALLHOME_URL" >> /tmp/debug
+      curl -skf -o /dev/null "$CALLHOME_URL" >/tmp/debug 2>&1
```

Comment:
Would it be helpful to add any other data here?  And related make this a `POST` rather than `GET`?  Something like:
```
(for x in /sys/class/net/*; do
  if ! -f $x/address; then continue; fi
  echo $x $(cat $x/address)
) > /tmp/net-devices.txt
curl -F coreos.inst=1,net-devices=@/tmp/net-devices -skf $CALLHOME_URL
```

or so?

Reply:
Cheers Colin! The `POST` is a great comment, we actually had `GET` for years and I've only fixed this few months ago so our endpoint actually accepts more correct `POST`. Let me fix that.

For the data, it's actually too late to provide any info to a provisioning management system because at this point everything was already feeded and initial configuration was done. At least in all our workflows, we heavily rely on configuration management systems like Ansible or Puppet which takes over on the initial boot and talks to our system in a different way (gathering facts and stuff).

However, it's a good point. Our endpoint actually accepts two calls - "finished" and "failed" when the latter accepts arbitrary text (log file) which is stored for later use. However this would require providing two URLs - one for success and one for failed case. This would probably be too complex. We were doing fine with just a simple "call home we are done" call, errors can be always investigated later on.

I am going to change the call to `POST` then.

Reply:
Rebased, how about this? Sending the `/tmp/debug/` file in the POST body. I can modify our "success" endpoint to store it in the database or at least log it.